code/game/objects/items/stacks/stack.dm

```diff
@@ -216,24 +216,29 @@
 	return get_amount() >= used
 
 /obj/item/stack/create_matter()
-	matter_per_piece = matter?.Copy() // this is used for refreshing matter amount in update_matter()
+	//Preserve matter_per_piece if already set, so it can be set manually.
+	if(!matter_per_piece)
+		matter_per_piece = matter?.Copy() // this is used for refreshing matter amount in update_matter()
 	if(istype(material))
 		LAZYINITLIST(matter_per_piece)
 		matter_per_piece[material.type] = max(matter_per_piece[material.type], round(MATTER_AMOUNT_PRIMARY * matter_multiplier))
-	. = ..()
+	//Don't call the parent, we handle matter completely differently, and it does a lot of stuff we'd just be overwriting completely
+	update_matter()
 
 /obj/item/stack/proc/update_matter()
-	matter = list()
+	//Preserve the matter contents for stacks that have extra matter not in matter_per_piece
```

Comment:
I'd argue that maybe always clearing the entire list of matter probably should only be done in the ``/obj/item/stack/material`` subclass instead? Since only material stacks are just pure material stacks unlike the other stacks. 
Things under ``/obj/item/stack`` tend to be items with a limited usage count, and it's not too far fetched to think it might be desirable to leave a trash item or something similar after it's used up. Basically like the wrapping paper and its carton tube. It definitely could find use in a number of ``/obj/item/stack`` like nanopaste, trauma kits packages etc.. 

Or maybe I could just add a list of matter that would be always added to the recomputed matter list? The reason I reused the matter list was that it was pretty straightforward to only overwrite the specific matter that was in matter per piece, and I don't think material stacks can/should change material at runtime.

Reply:
You can't split wrapping paper stacks, so that's not an issue.

Also, the problem here is that not accounting for all the matter that "comes with" an item means we're creating new matter out of thin air, which can be abused. I don't know what's so confusing about accounting for the entire matter cost of the thing? 

Also, the point is exactly that when you recycle the object you get the entire matter value so it stays consistent. Food also has a similar issue, though its not as bad, since the recipes don't cost matter, and a lot of those leaving trash are things you can't craft.

Making sure no new matter is being created out of thin air is pretty important if you want to have any sort of sensible economy.

Reply:
What would you do then? 
Handling that strictly in the wrapping paper would mean a whole lot of copy-pasta, which really makes that option kind of gross. That could be avoided by defining another list in the base stack class that would how a static matter amount to be added to the matter list when its rebuilt, but you said you're against that.

Reply:
I think I misunderstood what you meant then?
I thought you were completely against stacks having matter that was added on top of what the stack calculated already.